doc/src/manual/environment-variables.md

```diff
@@ -0,0 +1,327 @@
+# Environment Variables
+
+Julia may be configured with a number of environment variables, either in the
+usual way of the operating system, or in a portable way from within Julia.
+Suppose you want to set the environment variable [`JULIA_EDITOR`](@ref) to
+`vim`, then either type `ENV["JULIA_EDITOR"] = "vim"` for instance in the REPL
+to make this change on a case by case basis, or add the same to the user
+configuration file `.juliarc.jl` in the user's home directory to have
+a permanent effect.  The current value of the same environment variable is
+determined by evaluating `ENV["JULIA_EDITOR"]`.
+
+The environment variables that Julia uses generally start with `JULIA`. If
+[`Base.versioninfo`](@ref) is called with `verbose` equal to `true`, then the
+output will list all environment variables for which `JULIA` appears in the
+name.
+
+## File locations
+
+### `JULIA_HOME`
+
+The absolute path of the directory containing the Julia executable, which sets
+the global variable [`Base.JULIA_HOME`](@ref). If `$JULIA_HOME` is not set, then
+`Base.JULIA_HOME` defaults to its compile-time value.
+
+The executable itself is one of
+
+```
+$JULIA_HOME/julia
+$JULIA_HOME/julia-debug
+```
+
+by default.
+
+The global variable `Base.DATAROOTDIR` determines a relative path from
+`Base.JULIA_HOME` to the data directory associated with Julia. Then the path
+
+```
+$JULIA_HOME/$DATAROOTDIR/julia/base
+```
+
+determines the directory in which Julia initially searches for source files (via
+`Base.find_source_file()`).
+
+Likewise, the global variable `Base.SYSCONFDIR` determines a relative path to the
+configuration file directory. Then Julia searches for a `juliarc.jl` file at
+
+```
+$JULIA_HOME/$SYSCONFDIR/julia/juliarc.jl
+$JULIA_HOME/../etc/julia/juliarc.jl
+```
+
+by default (via `Base.load_juliarc()`).
+
+For example, a Linux installation with a Julia executable located at
+`/bin/julia`, a `DATAROOTDIR` of `../share`, and a `SYSCONFDIR` of `../etc` will
+have `JULIA_HOME` set to `/bin`, a source-file search path of
+
+```
+/share/julia/base
+```
+
+and a global configuration search path of
+
+```
+/etc/julia/juliarc.jl
+```
+
+### `JULIA_LOAD_PATH`
+
+A separated list of absolute paths that are to be appended to the variable
+[`Base.LOAD_PATH`](@ref). (In Unix-like systems, the path separator is `:`; in Windows
+systems, the path separator is `;`.) The `Base.LOAD_PATH` variable is where
+[`Base.require`](@ref) and `Base.load_in_path()` look for code; it defaults to the absolute
+paths
+
+```
+$JULIA_HOME/../local/share/julia/site/v$(Base.VERSION.major).$(Base.VERSION.minor)
```

Comment:
we could leave off `Base.` from `VERSION` and `LOAD_PATH` since those are exported from Base